proposals/0292-package-registry-service.md

```diff
@@ -157,18 +157,14 @@ A valid package scope matches the following regular expression pattern:
 ```
 
 A package's *name* uniquely identifies a package in a scope.
-The maximum length of a package name is 128 characters.
+The maximum length of a package name is 100 characters.
 A valid package name matches the following regular expression pattern:
 
 ```regexp
-\A\p{XID_Start}\p{XID_Continue}{0,127}\z
+\A[a-zA-Z\d_.-]{1,100}\z
```

Comment:
Do we want to allow names starting/ending with non-alphanumeric character?

Reply:
That sounds reasonable. You think something closer to the scope rules?

```suggestion
\A[a-zA-Z\d](?:[a-zA-Z\d]|[_.-](?=[a-zA-Z\d])){0,99}\z
```

Reply:
> You think something closer to the scope rules?

Yeah

> I could go either way with a leading underscore, but think leading dot or dash would be problematic.

OK, let's wait for other people's input. I don't have strong opinion on leading character either, but I do think that names must contain at least one alphanumeric character.

Reply:
I'd lean towards the suggested change that doesn't allow leading, trailing or consecutive underscore, dot, or dash.

During one of the review periods for SE-0292 we got feedback from the community in support of being more friendly to non-English centric package names. This was in regard to the original proposal that was diacritic-insensitive. And now we're completely doing away with diacritics and letters other than a-z. Is that a concern here?